programs/compressed-token/src/process_transfer.rs

```diff
@@ -517,7 +485,7 @@ pub fn get_input_compressed_accounts_with_merkle_context_and_check_signer<const
         }
         let compressed_account = CompressedAccount {
             owner: crate::ID,
-            lamports: input_token_data.is_native.unwrap_or_default(),
+            lamports: 0,
```

Comment:
@ananas-block 

If I understand this correctly, if we set input account lamports always to 0, compressed token accounts with lamports are not possible. that would be very useful to have, though, because it allows users to define spam filters for token accounts. 
also any output token accounts that were to be created with lamports would lead to awkward situations when trying to spend them 

fix: allow passing an optional `native_amount` in `input_token_data.`

Reply:
sure I can add that